File: easy/final_prices_with_discount.go
package easy

func FinalPricesWithDiscount(prices []int) []int {
	result := make([]int, len(prices))
	copy(result, prices)
	for i := 0; i < len(prices); i++ {
		for j := i + 1; j < len(prices); j++ {
			if prices[j] <= prices[i] {
				result[i] = prices[i] - prices[j]
				break
			}
		}

	}
	return result
}

File: src/easy/final_prices_with_discount.rs
pub fn final_prices_with_discount(prices: Vec<i32>) -> Vec<i32> {
    let mut result = prices.clone();
    for i in 0..result.len() {
        for j in i + 1..result.len() {
            if result[j] <= prices[i] {
                result[i] -= prices[j];
                break;
            }
        }
    }
    result
}